### PR TITLE
#44077 Moves publish2 hooks here, adds support for template-based workflows.

### PR DESCRIPTION
**Pull request by josh-t**
Same basic changes as `tk-maya`. Adds collection of `tk-houdini-alembicnode` and `tk-houdini-mantranode` items. Those apps have been modified to provide information to the collector which prevents having to configure an output template path or write any new publish plugins. The collector simply queries the app for any nodes and asks for the output path. If the path (or any frames) exist, it will publish it just like the output was dragged/dropped into the publisher. This should be possible for any apps/outputs that were generated prior to publisher being launched.

I will mark the relevant changes for review.

### CODE REVIEW

File: hooks/tk-multi-publish2/basic/collector.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+        # keep track of collected nodes
+        self._collected_nodes = []
+
+        # remember if we collect any alembic/mantra nodes
+        self._alembic_nodes_collected = False
+        self._mantra_nodes_collected = False
+
+        # methods to collect tk alembic/mantra nodes if the app is installed
+        self.collect_tk_alembicnodes(item)
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/collector.py]: These are the new collection methods.

File: hooks/tk-multi-publish2/basic/collector.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+        # get the icon path to display for this item
+        icon_path = os.path.join(
+            self.disk_location,
+            os.pardir,
+            "icons",
+            "houdini.png"
+        )
+        session_item.set_icon_from_path(icon_path)
+
+        # if a work file template is defined, add it to the item properties so
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/collector.py]: New template path injection into item properties. Same as Maya.

File: hooks/tk-multi-publish2/basic/collector.py [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+        """
+        Creates items for known output nodes
+
+        :param parent_item: Parent Item instance
+        """
+
+        for node_category in _HOUDINI_OUTPUTS:
+            for node_type in _HOUDINI_OUTPUTS[node_category]:
+
+                if node_type == "alembic" and self._alembic_nodes_collected:
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/collector.py]: Added these two conditionals. If the tk alembic/mantra nodes were collected, don't collect regular outputs. This avoid attempting to publish a tk node that contains a regular houdini node. I think it also makes sense that if the tk nodes are in play, publishing regular mantra/alembic nodes may be discouraged? Curious what you think.

File: hooks/tk-multi-publish2/basic/collector.py [hunk 4]
```diff
@@ -0,0 +1,303 @@
...
+                        path,
+                        frame_sequence=True
+                    )
+
+                    # the item has been created. update the display name to
+                    # include the node path to make it clear to the user how it
+                    # was collected within the current session.
+                    item.name = "%s (%s)" % (item.name, node.path())
+
+    def collect_tk_alembicnodes(self, parent_item):
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/collector.py]: New collection method for tk alembic nodes. Relies on changes to the app to make this simpler.

File: hooks/tk-multi-publish2/basic/collector.py [hunk 5]
```diff
@@ -0,0 +1,303 @@
...
+                parent_item, out_path)
+
+            # the item has been created. update the display name to
+            # include the node path to make it clear to the user how it
+            # was collected within the current session.
+            item.name = "%s (%s)" % (item.name, node.path())
+
+            self._alembic_nodes_collected = True
+
+    def collect_tk_mantranodes(self, parent_item):
```
Comment (josh-t) [inline hooks/tk-multi-publish2/basic/collector.py]: New collection method for tk mantra nodes. Relies on changes to the app to make this simpler.

File: hooks/tk-multi-publish2/basic/collector.py (hunk 4 above)
Comment (manneohrstrom) [reply hooks/tk-multi-publish2/basic/collector.py]: docstrings pls :)

**Pull request merged**